static/app/views/releases/list/releaseAdoptionChart.tsx

```diff
@@ -0,0 +1,237 @@
+import Count from 'app/components/count';
+import {Panel, PanelBody, PanelFooter} from 'app/components/panels';
+import Placeholder from 'app/components/placeholder';
+import {t, tct} from 'app/locale';
+import space from 'app/styles/space';
+import {GlobalSelection, Organization, Project, Release} from 'app/types';
+import {ReactEchartsRef} from 'app/types/echarts';
+import withApi from 'app/utils/withApi';
+import {DisplayOption} from 'app/views/releases/list/utils';
+import {ReleaseHealthRequestRenderProps} from 'app/views/releases/utils/releaseHealthRequest';
+
+type Props = WithRouterProps & {
+  api: Client;
+  organization: Organization;
+  selection: GlobalSelection;
+  releases: Release[];
+  project: Project;
+  getHealthData: ReleaseHealthRequestRenderProps['getHealthData'];
+  activeDisplay: DisplayOption;
+  showPlaceholders: boolean;
+};
+
+type State = {
+  width: number;
+  height: number;
+};
+
+class ReleaseAdoptionChart extends React.PureComponent<Props, State> {
+  state = {
+    width: -1,
+    height: -1,
+  };
+
+  ref: null | ReactEchartsRef = null;
+
+  /**
+   * Syncs component state with the chart's width/heights
+   */
+  updateDimensions = () => {
+    const chartRef = this.ref?.getEchartsInstance?.();
+    if (!chartRef) {
+      return;
+    }
+
+    const width = chartRef.getWidth();
+    const height = chartRef.getHeight();
+    if (width !== this.state.width || height !== this.state.height) {
+      this.setState({
+        width,
+        height,
+      });
+    }
+  };
+
+  handleRef = (ref: ReactEchartsRef): void => {
+    if (ref && !this.ref) {
+      this.ref = ref;
+      this.updateDimensions();
+    }
+
+    if (!ref) {
+      this.ref = null;
+    }
+  };
+
+  renderEmpty() {
+    return (
+      <ChartPanel>
+        <ChartBody withPadding>
+          <ChartHeader>
+            <Placeholder height="24px" />
+          </ChartHeader>
+          <Placeholder height="200px" />
+        </ChartBody>
+        <ChartFooter>
+          <Placeholder height="24px" />
+        </ChartFooter>
+      </ChartPanel>
+    );
+  }
+
+  render() {
+    const {
+      showPlaceholders,
+      releases,
+      project,
+      activeDisplay,
+      router,
+      selection,
+      getHealthData,
+    } = this.props;
+    const {start, end, period, utc} = selection.datetime;
+
+    if (showPlaceholders) {
+      return this.renderEmpty();
+    }
+
+    const get24hCountByProject = getHealthData.get24hCountByProject(
+      Number(project.id),
+      activeDisplay
+    );
+
+    const releasesSeries = releases.map(release => {
+      const releaseVersion = release.version;
+
+      const timeSeries = getHealthData.getTimeSeries(
+        releaseVersion,
+        Number(project.id),
+        activeDisplay
+      );
+
+      return {
+        data: timeSeries[0].data.map((d, i) => ({
+          name: d.name,
+          value:
+            d.value > 0 && timeSeries[1].data[i].value > 0
```

Comment:
this bit is a little hard to read is it possible to have some aliases for `timeSeries[0].data` and `timeSeries[1].data` maybe like
`const denom = timeSeries[1].data`, feel free to replace `denom` with something more explicit

Reply:
One is in release other is total data, I can name them that, it'll be more explicit